applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -621,364 +866,157 @@ namespace Kratos
+		vector_norm = MathUtils<double>::Norm(delta_x);
+		if (vector_norm > numerical_limit) delta_x /= vector_norm;
 
-		double alpha = 0.0;
-		if (this->GetProperties().Has(RAYLEIGH_ALPHA))
-		{
-			alpha = this->GetProperties()[RAYLEIGH_ALPHA];
-		}
-		else if (rCurrentProcessInfo.Has(RAYLEIGH_ALPHA))
-		{
-			alpha = rCurrentProcessInfo[RAYLEIGH_ALPHA];
-		}
 
-		double beta = 0.0;
-		if (this->GetProperties().Has(RAYLEIGH_BETA))
-		{
-			beta = this->GetProperties()[RAYLEIGH_BETA];
-		}
-		else if (rCurrentProcessInfo.Has(RAYLEIGH_BETA))
-		{
-			beta = rCurrentProcessInfo[RAYLEIGH_BETA];
+		Bisectrix = rotated_nx0 + delta_x;
+		vector_norm = MathUtils<double>::Norm(Bisectrix);
+		if (vector_norm > numerical_limit) Bisectrix /= vector_norm;
+
+		bounded_matrix<double,msDimension,msDimension> n_xyz = ZeroMatrix(msDimension);
+		for (unsigned int i = 0; i < msDimension; ++i) {
+			n_xyz(i, 0) = -1.0 * rotated_coordinate_system(i, 0);
+			n_xyz(i, 1) = 1.0 * rotated_coordinate_system(i, 1);
+			n_xyz(i, 2) = 1.0 * rotated_coordinate_system(i, 2);
 		}
 
-		rDampingMatrix += alpha * mass_matrix;
-		rDampingMatrix += beta  * stiffness_matrix;
+		bounded_matrix<double,msDimension,msDimension> Identity = ZeroMatrix(msDimension);
+		for (unsigned int i = 0; i < msDimension; ++i) Identity(i, i) = 1.0;
+		Identity -= 2.0 * outer_prod(Bisectrix, Bisectrix);
+		n_xyz = prod(Identity, n_xyz);
 
+		this->mLocalRotationMatrix = n_xyz;
+		return n_xyz;
 		KRATOS_CATCH("")
 	}
 
 
-	bounded_vector<double,CrBeamElement3D2N::msElementSize> CrBeamElement3D2N::CalculateBodyForces()
-	{
-		KRATOS_TRY
-		//getting shapefunctionvalues for linear SF
-		const Matrix& Ncontainer = this->GetGeometry().ShapeFunctionsValues(
-			GeometryData::GI_GAUSS_1);
 
-		bounded_vector<double,msDimension> equivalent_line_load = ZeroVector(msDimension);
-		bounded_vector<double,msElementSize> body_forces_global = ZeroVector(msElementSize);
 
-		const double A = this->GetProperties()[CROSS_AREA];
-		const double l = this->CalculateCurrentLength();
-		const double rho = this->GetProperties()[DENSITY];
+	Vector CrBeamElement3D2N::CalculateSymmetricDeformationMode(const Vector& VectorDifference)
+	{
+		Vector phi_s = ZeroVector(msDimension);
+		if(this->mIterationCount!=0)
+		{phi_s = prod(Matrix(trans(this->mLocalRotationMatrix)), VectorDifference);
+		phi_s *= 4.00;}
+		return phi_s;
+	}
 
-		//calculating equivalent line load
-		for (int i = 0; i < msNumberOfNodes; ++i)
-		{
-			equivalent_line_load += A * rho*
-				this->GetGeometry()[i].
-				FastGetSolutionStepValue(VOLUME_ACCELERATION)*Ncontainer(0, i);
-		}
+	Vector CrBeamElement3D2N::CalculateAntiSymmetricDeformationMode(const Vector& Bisectrix)
+	{
+		Vector phi_a = ZeroVector(msDimension);
+		if(this->mIterationCount!=0)
+		{Vector rotated_nx0 = ZeroVector(msDimension);
+		for (unsigned int i = 0; i < msDimension; ++i) rotated_nx0[i] = this->mLocalRotationMatrix(i, 0);
+		Vector temp_vector = ZeroVector(msDimension);

```

Comment:
what do u need this zero initialization for?

Reply:
as far as I understand this is the vector which will contain the result of the crossproduct calculation